languages/python/django-oso/django_oso/partial.py

```diff
@@ -91,17 +91,23 @@ def and_expr(expr: Expression, model: Model, **kwargs):
     return q
 
 
-def compare_expr(expr: Expression, _model: Model, path=(), **kwargs):
+def compare_expr(expr: Expression, model: Model, path=(), **kwargs):
     q = Q()
     (left, right) = expr.args
     left_path = dot_path(left)
     if left_path:
         return COMPARISONS[expr.operator](q, "__".join(path + left_path), right)
     else:
-        if isinstance(right, Model):
+        if isinstance(right, model):
             right = right.pk
         else:
+            # QUESTION: Should this just return an empty set? Similar to how invalid isa
+            # returns empty set.
             raise UnsupportedError(f"Unsupported comparison: {expr}")
+
+        if not expr.operator in ('Eq', 'Unify'):
+            raise UnsupportedError(f"Comparing model only supported with Eq or Unify not: {expr}")
```

Comment:
Why is this necessary?

Reply:
Because it doesn't make sense otherwise.  If you write `this < post`, you would probably expect to use the natural ordering of post, which isn't something we know about, not to get all posts that have a primary key less than that of the right hand side.

Reply:
Cool, can we use the error message from above (`Unsupported comparison: {expr}`) once we replace that with the `Q(pk__in=[])` thing?

Reply:
I used a different message because it is more specific and tells you that a Unification or Equality comparison would be valid, not just that the comparison is unsupported.

Reply:
Revised to match other error more closely.